### PR TITLE
💅 integrate prettier and eslint

### LINKED ISSUE #1235: Move to prettier & eslint
This will increase developer happiness and speed up productivity both in development and code review!

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
closes #1235

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "parser": "typescript",
+  "trailingComma": "es5",
+  "singleQuote": true
+}
```
Comment (a-b-r-o-w-n) [inline .prettierrc]: We can debate here on the prettier formatting options. https://prettier.io/docs/en/options.html

File: package.json [hunk 2]
```diff
@@ -35,6 +40,17 @@
...
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{ts,tsx,js,jsx}": [
+      "prettier --write",
+      "git add"
+    ]
+  },
```
Comment (a-b-r-o-w-n) [inline package.json]: We can potentially expand this to other file types, including markdown and css

**Comment by cwhitten** [comment]
For any observers wondering why eslint instead of tslint, see [this issue on the TypeScript project](https://github.com/Microsoft/TypeScript/issues/29288)

**Comment by a-b-r-o-w-n** [comment]
@tonyanziano and I both did a regression pass which uncovered two unrelated issues (related to #1219 ). Merging this and will file new issues against the regressions.

**Pull request merged**